File: Medium/2161_Partition_Array_According_to_Given_Pivot/Solution.py
from typing import List, Optional


class Solution:
    def pivotArray(self, nums: List[int], pivot: int) -> List[int]:
        smaller = []
        equal = []
        greater = []

        for num in nums:
            if num < pivot:
                smaller.append(num)
            elif num == pivot:
                equal.append(num)
            else:
                greater.append(num)

        return smaller + equal + greater


if __name__ == "__main__":
    solution = Solution()

    # test cases 1
    nums1 = [9, 12, 5, 10, 14, 3, 10]
    pivot1 = 10
    print(solution.pivotArray(nums1, pivot1))  # Output: [9, 5, 3, 10, 10, 12, 14]

    # test cases 2
    nums2 = [-3, 4, 3, 2]
    pivot2 = 2
    print(solution.pivotArray(nums2, pivot2))  # Output: [-3, 2, 4, 3]

File: Medium/2161_Partition_Array_According_to_Given_Pivot/Solution.java
import java.util.ArrayList;
import java.util.List;


public class Solution {
    public int[] pivotArray(int[] nums, int pivot) {
        List<Integer> smaller = new ArrayList<>();
        List<Integer> equal = new ArrayList<>();
        List<Integer> greater = new ArrayList<>();

        for (int num : nums) {
            if (num < pivot) {
                smaller.add(num);
            } else if (num == pivot) {
                equal.add(num);
            } else {
                greater.add(num);
            }
        }

        int[] result = new int[smaller.size() + equal.size() + greater.size()];
        int index = 0;

        for (int num : smaller) {
            result[index++] = num;
        }
        for (int num : equal) {
            result[index++] = num;
        }
        for (int num : greater) {
            result[index++] = num;
        }

        return result;
    }

    public static void main(String[] args) {
        Solution solution = new Solution();

        // test cases 1
        int[] nums1 = {9, 12, 5, 10, 14, 3, 10};
        int pivot1 = 10;
        int[] result1 = solution.pivotArray(nums1, pivot1);
        for (int num : result1) {
            System.out.print(num + " ");
        }
        System.out.println();

        // test cases 2
        int [] nums2 = {-3, 4, 3, 2};
        int pivot2 = 2;
        int[] result2 = solution.pivotArray(nums2, pivot2);
        for (int num : result2) {
            System.out.print(num + " ");
        }
        System.out.println();
        
    }
}